ext/curl/interface.c

```diff
@@ -104,7 +104,7 @@ static int php_curl_option_str(php_curl *ch, zend_long option, const char *str,
 	CURLcode error = CURLE_OK;
 
 	if (strlen(str) != len) {
-		php_error_docref(NULL, E_WARNING, "Curl option contains invalid characters (\\0)");
+		zend_type_error("%s(): cURL option cannot contain any null-bytes", get_active_function_name());
```

Comment:
Maybe it would be worth to introduce a new function that adds the current function/method name to the beginning. It might be useful if we start adding OO interfaces.